test/conftest.py

```diff
@@ -95,6 +104,15 @@ def pytest_collection_modifyitems(items):
         elif "weaviate" in item.nodeid:
             item.add_marker(pytest.mark.weaviate)
 
+        # if the cli argument "--document_store_type" is used, we want to skip all tests that have markers of other docstores
+        # Example: pytest -v test_document_store.py --document_store_type="memory" => skip all tests marked with "elasticsearch"
+        document_store_types_to_run = config.getoption("--document_store_type")
+        for cur_doc_store in ["elasticsearch", "faiss", "sql", "memory", "milvus", "weaviate"]:
+            if cur_doc_store in item.keywords and cur_doc_store not in document_store_types_to_run:
+                skip_docstore = pytest.mark.skip(
+                    reason=f'{cur_doc_store} is disabled. Enable via pytest --document_store_type="{cur_doc_store}"')
+                item.add_marker(skip_docstore)
+
 
 @pytest.fixture(scope="session")
 def elasticsearch_fixture():
```

Comment:
Why we need these now?. Pytest not going to call this code.

Reply:
Yep, most of them are not used anymore (except elasticsearch_fixture). 
See bullet point one in the "limitations / future work" of the PR desc .
Would like to tackle this in a separate PR though as this one is becoming quite big already